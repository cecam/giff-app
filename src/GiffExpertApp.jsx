import { useState } from "react";

import AddCategory from './components/AddCategory'
import GiffGrid from './components/GiffGrid'

const GiffExpertApp = ({defaultCategories = []}) => {
    const [categories, setCategories] = useState(defaultCategories)

    return ( 
        <>
            <h2>GiffExpertApp</h2>
            <AddCategory setCategories={setCategories} />
            <hr />

            <ol>
                {
                    categories.map(category => (
                        <GiffGrid 
                            category={category} 
                            key={category} 
                        />
                    ))
                }                
            </ol>
        </>
     );
}
 
export default GiffExpertApp;