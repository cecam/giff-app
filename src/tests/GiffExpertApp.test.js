import React from 'react';
import { shallow } from 'enzyme';

import GiffExpertApp from '../GiffExpertApp'

describe('Tests on <GiffExpertApp />', () => {

    test('should print the component', () => {
        const wrapper = shallow( <GiffExpertApp /> )

        expect( wrapper ).toMatchSnapshot()
    });

    test('should print a list of categories', () => {
        const categories = ['naruto', 'one punch']
        const wrapper = shallow( <GiffExpertApp defaultCategories={categories} /> )

        expect( wrapper ).toMatchSnapshot()
        expect( wrapper.find('GiffGrid').length ).toBe(categories.length)

    });
})
