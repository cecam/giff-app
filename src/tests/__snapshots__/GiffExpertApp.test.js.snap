// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <GiffExpertApp /> should print a list of categories 1`] = `
<Fragment>
  <h2>
    GiffExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GiffGrid
      category="naruto"
      key="naruto"
    />
    <GiffGrid
      category="one punch"
      key="one punch"
    />
  </ol>
</Fragment>
`;

exports[`Tests on <GiffExpertApp /> should print the component 1`] = `
<Fragment>
  <h2>
    GiffExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
